### PR TITLE
TFSNOW-10: Creating __AMAZON_ORDERS_USG usage role

### PR DESCRIPTION
**Pull request by NagasairamM**
Please review and approve the PR

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -45,4 +45,16 @@ resource "snowflake_grant_ownership" "schema_grant" {
     object_type = "SCHEMA"
     object_name = "\"${snowflake_database.db.name}\".\"${snowflake_schema.schema.name}\""
   }
+}
+resource "snowflake_role" "role1" {
```
Comment (csarat) [inline main.tf:49]: Give proper names to the resources

**Review by csarat — CHANGES_REQUESTED** [review]
Give proper names to all the terraform resource names

**Comment by csarat** [comment]
Give proper title to the PR. `<Jira#>: <overall description of the PR>` for example for this PR `TFSNOW-10: Creating __AMAZON_ORDERS_USG usage role`. This can still be changed.

Commit message should be description of that commit with the format `<Jira#>: <message>` for example `TFSNOW-10: Creating usage role`. This is FYI. you cannot change this anymore.

**Comment by NagasairamM** [comment]
Creating __AMAZON_ORDERS_USG usage role;
Based on the given suggestions the code has been changed and raised request again. please review and approve the PR

File: main.tf [hunk 2]
```diff
@@ -11,11 +11,11 @@ provider "snowflake" {
   role = "SYSADMIN"
 }
 
-resource "snowflake_database" "db" {
+resource "snowflake_database" "amazon_db1" {
```
Comment (csarat) [inline main.tf:14]: This is much better. `amazon_db` makes more sense. 
Why is `1` required here?

File: main.tf [hunk 3]
```diff
@@ -28,21 +28,33 @@ resource "snowflake_schema" "schema" {
...
     object_type = "SCHEMA"
-    object_name = "\"${snowflake_database.db.name}\".\"${snowflake_schema.schema.name}\""
+    object_name = "\"${snowflake_database.amazon_db1.name}\".\"${snowflake_schema.orders_schema1.name}\""
+  }
+}
+resource "snowflake_role" "amazon_role2" {
+  provider = snowflake.security_admin
+  name     = "__AMAZON_ORDERS_USG"
+}
+resource "snowflake_grant_privileges_to_account_role" "schema_grant" {
```
Comment (csarat) [inline main.tf:53]: can you come up with a better name instead of `schema_grant`

**Review by csarat — CHANGES_REQUESTED** [review]
Check my comments

**Comment by NagasairamM** [comment]
Update the code according to the given preferences! please review and approve the PR.

File: main.tf (hunk 3 above)
Comment (NagasairamM) [reply main.tf:61]: I think schema_grant word makes more sense.

File: main.tf (hunk 2 above)
Comment (NagasairamM) [reply main.tf]: please find the new updated with name.

**Comment by NagasairamM** [comment]
The code was updated based on the given suggestions, Please review the code.

**Comment by NagasairamM** [comment]
Please review the code.

**Review by csarat — APPROVED** [review]

**Pull request merged**